streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryWindowStore.java

```diff
@@ -135,6 +135,9 @@ public void put(final Bytes key, final byte[] value, final long windowStartTimes
             } else {
                 segmentMap.computeIfPresent(windowStartTimestamp, (t, kvMap) -> {
                     kvMap.remove(keyBytes);
+                    if (kvMap.isEmpty()) {
```

Comment:
Nice catch. Though I'm curious how the unit test could expose this? :)

Reply:
This was just something I thought of to improve